EIPS/eip-5269.md

```diff
@@ -0,0 +1,102 @@
+1. Create a lower entry barrier of detecting interfaces for Dapp developer and Smart Contract developers.
+2. Allow detecting different interfaces for different callers.
+3. Potentially making it easier to work with ENS [EIP-137](./eip-137.md). For example, there was a proposal to record interface information on ENS. Having a human readable interface provides an easier way for smart contract to create such ENS domains easily.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+1. Any compliant smart contract MUST implement the following interface
+
+```solidity
+interface IERC5269 {
+  function supportsHumanReadableInterface(uint256 ercNumber, address caller) external view returns (boolean isSupported);
+}
+```
+
+For example any smart contract that is an `IERC5629` MUST return `true` for input `{ercNumber: 5269, caller: <any>}`
+
+```solidity
+contract DemoContract is IERC5269 {
+  function supportsHumanReadableInterface(uint256 ercNumber, address caller) external view returns (boolean isSupported) {
+    if (ercNumber === 5269) {
+      return true;
+    } 
+    return false;
+}
+```
+
+2. Unless superseded by future EIPs, the `ercNumber` for any input greater than 2**32 = `0x100000000` in hex or `4294967296` in decimal MUST return `false` and are reserved for other type of interfaces. 
+
+3. Optionally, implementing smart contract MAY implement the following enumerable inetrface
+
+```solidity
+interface IERC5269Enumberable is IERC5269 {
+  function supportsTotalHumanReadableInterfaces(address caller) external view returns (uint256 numberOfInterface);
+  function supportsHumanReadableInterfaceByIndex(uint256 index, address caller) external view returns (uint256 ercNumber);
+}
+```
+
+For example an [EIP-721](./eip-721.md) compliant contract MAY be implemented as below
+
+```solidity
+interface ERC721 is IERC5269Enumberable {
+  function supportsTotalHumanReadableInterfaces(address caller) external view returns (uint256 numberOfInterface) {
+    return 3;
+  }
+  function supportsHumanReadableInterfaceByIndex(uint256 index, address caller) external view returns (uint256 ercNumber) {
+    require(index < 3);
+    if (index == 0) {
+      return 20;
+    } else if (index == 1) { 
+      return 165;
+    } else if (index == 2) {
+      return 5269; 
+    }
+}
+```
+
+
+## Rationale
+
+1. We simply return the eip numbers in an array reflecting the practice that an smart contract usually resemble more than one interface.
+
+2. We didn't require the ordering of return value. And we only suggest but didn't require deduplication because it's cheaper to do such computation outside of chain.
+
+3. Compared to [EIP-165](./eip-165.md), we also add an addition input of `address caller`, given the increasing popularity of proxy patterns such as those enabled by [EIP-1967](./eip-1967.md).
```

Comment:
I'd include an answer to the question "why not use `msg.sender`?" in the rationale section.

Reply:
Thank you, that's a good way to answer it. Added a sentence